Feature dashboard stock

lomamech opened the pull request:
- Add indicator panel
- Addition of a loading when downloading data
- Complete panel for Dashbord
- Add link to registry of stock or lot

lomamech committed:
Fix and resolve conflict

lomamech committed:
Fix and resolve conflict

lomamech committed:
improvement(DashbordStock)
- Clean and sanitaze code

closes #4856

mbayopanda commented on client/src/css/structure.css:
```diff
@@ -888,19 +888,35 @@ a[disabled] {
 }
 
+.icon-expired {
+  color: #ff000083 !important;
```
- Is it necessary to add `!important`?
- Why using the HEX8 (#XXXXXXXX) color notation in place of the well-known HEX color (#XXXXXX)?

lomamech replied:
I discovered it for the first time in this file, it is in several classes, this will force that the defined values have priority over any other value that will be defined

jmcameron replied:
The last two hex digits in the long format control the transparency of the item.  So '83' means Transparency should be 83/256 (or 32% transparent).

jmcameron replied:
See https://css-tricks.com/8-digit-hex-codes/

mbayopanda commented on server/controllers/stock/index.js:
```diff
@@ -591,6 +594,167 @@ function listMovements(req, res, next) {
+    WHERE p.user_id = ?
+    ORDER BY d.text ASC`;
+
+  db.exec(getDepotsByUser, [req.session.user.id])
```
Interesting

jniles replied:
I like this.  In the long run, I can see this code being put in the users controller, then we can have `/users/:id/depots` as a route to get the depots associated with that user.

mbayopanda commented on client/src/modules/stock/dashboard/dashboard.js:
```diff
@@ -0,0 +1,20 @@
+  // fired on startup
+  function startup() {
+
+  }
```
?

lomamech replied:
See that everything is done with the component, there is unfortunately nothing inside the model

mbayopanda commented on client/src/modules/stock/dashboard/dashboard.html:
```diff
@@ -0,0 +1,74 @@
+    </div>
+
+    <div class="row">
+      <div class="col-md-6">
```
Est-ce possible d'essayer avec `col-md-4` pour avoir trois colonnes de cette maniere on pourrait gagner en hauteur

lomamech replied:
Let me try

lomamech replied:
![image](https://user-images.githubusercontent.com/8035588/99365644-db66b680-28b7-11eb-9f81-3aa9c9c3baea.png)

jmcameron replied:
Thanks for including this image.  I did not run it with  the IMCK database, so it was helpful to see this tableau.

mbayopanda reviewed (commented):
The dashboard looks nice 👍 

I left some comments

jmcameron reviewed (commented):
I like the concept.  However, I suggest that you actually list the items out on the Dashboard (at least a limited number).  Having the link to see the details is great, but showing some abbreviated details here would be good.
![image](https://user-images.githubusercontent.com/62145/99309747-c2ee9180-280e-11eb-954e-d2bc2c2d01ac.png)

Note that the "Stock Exit" page has similar lists which could be adapted for this dashboard.

I glanced over the code and it looks okay.

lomamech committed:
Fix and solve conflict with branch master

lomamech committed:
Fix and solve conflict

jniles commented on client/src/js/components/bhDashboardStock.js:
```diff
@@ -0,0 +1,109 @@
+  .component('bhDashboardStock', {
+    templateUrl : 'modules/templates/bhDashboardStock.tmpl.html',
+    controller  : DashboardStock,
+    transclude  : true,
```
Do you use transclude?  It doesn't look like it.  Could you remove this?

jniles commented on client/src/js/components/bhDashboardStock.js:
```diff
@@ -0,0 +1,109 @@
+function DashboardStock(StockDashBoard, Notify) {
+  const $ctrl = this;
+
+  $ctrl.$onInit = function onInit() {
```
Hmm... instead of having this parsing, it probably makes better sense to break up a lot of these into separate, small components.  Like `bh-stock-panel-expired` and `bh-stock-panel-out-of-stock`.  That way, if there is a bug in one of these, we have an easier time fixing it and do not have to parse through a ton of other code.

lomamech replied:
no problem

jniles commented on client/src/js/components/bhDashboardStock.js:
```diff
@@ -0,0 +1,109 @@
+    },
+  });
+
+DashboardStock.$inject = [
```
Can you add the word "Controller" to this?

jniles commented on client/src/js/components/bhDashboardStock.js:
```diff
@@ -0,0 +1,109 @@
+
+    }
+
+    StockDashBoard.read({ status : $ctrl.status })
```
Let's call this "StockDashboard", since Dashboard is one word.

jniles commented on client/src/modules/stock/dashboard/dashboard.routes.js:
```diff
@@ -0,0 +1,9 @@
+    $stateProvider
+      .state('/stock/dashboard', {
+        url : '/stock/dashboard',
+        controller : 'StockDashBoardController as StockDashBoardCtrl',
```
You don't need to have a controller for each view.  It looks like you invented a controller just to put it here.  If the view doesn't have a controller just put in the templateUrl and that should be fine.

jniles reviewed (commented):
Just a few comments here.

lomamech committed:
Clean and sanitaze code

lomamech committed:
improvement: using HEX color instead HEX8

lomamech committed:
Feature: Use new component for stock expired

lomamech committed:
improvement(Dashboard stock)
- Remove params transclude
- Use separate small components for all panel
- Remove component bhDashboardstock

lomamech committed:
Solve conflict

jniles reviewed (approved):
LGTM.

bors r+
